plugins/groups_manager.py

```diff
@@ -21,14 +21,54 @@
 from __future__ import absolute_import
 from __future__ import unicode_literals
 
+import argparse
 import libcomps
+import re
 
 from dnfpluginscore import _, logger
-from dnf.cli.option_parser import OptionParser
 import dnf
 import dnf.cli
 
 
+RE_GROUP_ID_VALID = "-a-z0-9_.:"
+RE_GROUP_ID = re.compile(r"^[{}]+$".format(RE_GROUP_ID_VALID))
+RE_LANG = re.compile(r"^[-a-zA-Z0-9_.@]+$")
+COMPS_XML_OPTIONS = {
+    'default_explicit': True,
+    'uservisible_explicit': True,
+    'empty_groups': True}
+
+
+def group_id_type(value):
+    '''group id validator'''
+    if not RE_GROUP_ID.match(value):
+        raise argparse.ArgumentTypeError(_('Invalid group id'))
+    return value
+
+
+def translation_type(value):
+    '''translated texts validator'''
+    data = value.split(':', 2)
+    if len(data) != 2:
+        raise argparse.ArgumentTypeError(
+            _("Invalid translated data, should be in form 'lang:text'"))
+    lang, text = data
+    if not RE_LANG.match(lang):
+        raise argparse.ArgumentTypeError(_("Invalid/empty language for translated data"))
+    return lang, text
+
+
+def text_to_id(text):
+    '''generate group id based on its name'''
+    group_id = text.lower()
+    group_id = re.sub('[^{}]'.format(RE_GROUP_ID_VALID), '', group_id)
+    if not group_id:
+        raise dnf.cli.CliError(
+            _("Can't generate group id from '{}'. Please specify group id using --id.").format(
+                text))
+    return group_id
```

Comment:
I know I'm complicating things (so feel free to push back - e.g. regexps will likely be a problem...), but I think this kind of validation belongs to libcomps to be possibly shared among projects.

Reply:
This check is meant for this situation:
`dnf groups-manager --name=ěščř`
In case the id of newly added group is not passed using `--id` command line argument the plugin generates one based on the name. The generator is quite simple (`text_to_id()` function) and it just converts name to lower case and removes characters that are not allowed in the id. And this might fail in case there are no characters left after the removal.